onnxruntime/core/optimizer/matmul_transpose_fusion.cc

```diff
@@ -31,7 +36,7 @@ static std::pair<bool, Node*> IsInputTranspose(const Graph& graph, NodeArg& node
   }
 
   if (is_trans_on_last_two_dims) {
-    is_trans_on_last_two_dims = (int64_t)perms[rank - 2] == rank - 1 && (int64_t)perms[rank - 1] == rank - 2;
+    is_trans_on_last_two_dims = perms[rank - 2] == rank - 1 && perms[rank - 1] == rank - 2;
   }
 
   if (!is_trans_on_last_two_dims) {
```

Comment:
there's a const_cast on line 46. GitHub doesn't let me comment on it, perhaps it's too far out of the diff :). i know it's not related to an existing PR comment, but it is unnecessary ugliness - can we just have this function take a non-const Graph& instead, since it returns a non-const Node*?

Reply:
if we do that, maybe this function can be renamed to GetTransposeNodeFromOutput() or something. also it looks like result.second is non-null iff result.first is true. maybe just return a pointer then?